src/plugins/kibana/public/dashboard/components/panel/lib/visualization.js

```diff
@@ -16,6 +16,7 @@ export default function visualizationLoader(savedVisualizations, Private) { // I
         savedObj: savedVis,
         panel: panel,
         uiState: savedVis.uiStateJSON ? JSON.parse(savedVis.uiStateJSON) : {},
+        setUiState(uiState) { savedVis.vis.setUiState(uiState); },
```

Comment:
I'm not a fan of there being a setter here. There are other properties that are consumed off of the passed in `$scope`, why not do the same with the uiState as well?